int8 action_type
  int8 GET_INFO = 0
  int8 ADD_GROUP = 1
  int8 REMOVE_GROUP = 2
  int8 SET_MARKER_POSE = 3
  int8 SET_SHOW_PATH = 4
  int8 SET_PLAN_ON_MOVE = 5
  int8 SET_EXECUTE_ON_MOVE = 6
  int8 SET_POSITION_TOLERANCE = 7
  int8 SET_ORIENTATION_TOLERANCE = 8
  int8 SET_JOINT_MAP = 9
  int8 EXECUTE_STORED_POSE = 10
  int8 TOGGLE_POSTURE_CONTROLS = 11
  int8 SYNC_MARKERS_TO_ROBOT = 12
  int8 PLAN_TO_MARKER = 13
  int8 EXECUTE_PLAN = 14
  int8 ADD_NAVIGATION_WAYPOINT = 15
  int8 DELETE_NAVIGATION_WAYPOINT = 16
  int8 PLAN_NAVIGATION_PATH = 17
  int8 EXECUTE_NAVIGATION_PATH = 18

# name of groups to add, remove, or index into for below structures (i.e., the order of this list defines the data below)
string[] group_name

# types of groups if being added
string[] group_type

# should be "all_points" or "last_point"
string[] stored_pose_name
string[] path_visualization_mode

# sets the interactive markers (for cartesian groups only) to the given poses
geometry_msgs/PoseStamped[] goal_pose

JointMask[] joint_mask

ToleranceInfo[] tolerance

# the names of the navigation_points that will be deleted
string[] navigation_waypoint_name

# current setting of flags
bool[] execute_on_plan
bool[] plan_on_move

----

# the action type
int8 action_type

# returns all groups in robot config file
string[] group_name
string[] group_type

# returns the subset of group_name that are displayed/active
string[] active_group_name

# returns the names of groups that have valid plans generated 
bool[] plan_found 

# current joint mask for each group in active_group_name
JointMask[] joint_mask

# current joint names for each group
JointNameMap[] joint_names

# current path visualization modes
string[] path_visualization_mode

# current list of (set) tolerance
ToleranceInfo[] tolerance

# current tolerance settings
ToleranceInfo[] tolerance_setting

# current setting of flags
bool[] execute_on_plan
bool[] plan_on_move